src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -942,3 +1137,40 @@ def _show_command(self, name, getter_name='get', getter_type=None, custom_comman
         getter_op = self._resolve_operation(merged_kwargs, getter_name, getter_type, custom_command=custom_command)
         _cli_show_command(self.command_loader, '{} {}'.format(self.group_name, name), getter_op=getter_op,
                           custom_command=custom_command, **merged_kwargs)
+
+
+def register_cache_arguments(cli_ctx):
+    from knack import events
+    from azure.cli.core.commands.parameters import CaseInsensitiveList
+
+    cache_dest = '_cache'
+
+    def add_cache_arguments(_, **kwargs):  # pylint: disable=unused-argument
+
+        command_table = kwargs.get('commands_loader').command_table
+
+        if not command_table:
+            return
+
+        class CacheAction(argparse.Action):  # pylint:disable=too-few-public-methods
+
+            def __call__(self, parser, namespace, values, option_string=None):
+                setattr(namespace, cache_dest, values)
+                # save caching status to CLI context
+                cmd = getattr(namespace, 'cmd', None) or getattr(namespace, '_cmd', None)
+                cmd.cli_ctx.data[cache_dest] = values
+
+        for command in command_table.values():
```

Comment:
The should all be registered in the same place (and this is registered in the same place as similar methods). Where they are defined depends on where they "fit".  This has nothing to do with ARM, so it would not make sense to put in the arm package with register_ids_arguments.

Reply:
I don't dispute that this is a good place to put the argument. I was just wondering if, in the future, a unified way / place for registering arguments outside of command modules would be helpful...